### ISSUE TITLE
[Issue]: AMD Clang - Warnings

### ISSUE THREAD

**Issue by kiritigowda** [ROCm/MIVisionX#1455]
### Problem Description

Switch to AMD Clang++ creates a few warnings

```
[  7%] Building CXX object amd_openvx/openvx/CMakeFiles/openvx.dir/ago/ago_haf_cpu_color_convert.cpp.o
/home/kiriti/develop/mivisionx-kiriti/MIVisionX/amd_openvx/openvx/ago/ago_drama_alloc.cpp:771:141: warning: result of comparison of constant 2 with expression of type 'vx_direction_e' is always false [-Wtautological-constant-out-of-range-compare]
  771 |                     if (it == superNodeInfo->inputList.end() && (node->parameters[i].direction == VX_INPUT || node->parameters[i].direction == VX_BIDIRECTIONAL))
      |                                                                                                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~
/home/kiriti/develop/mivisionx-kiriti/MIVisionX/amd_openvx/openvx/ago/ago_drama_alloc.cpp:774:143: warning: result of comparison of constant 2 with expression of type 'vx_direction_e' is always false [-Wtautological-constant-out-of-range-compare]
  774 |                     if (it == superNodeInfo->outputList.end() && (node->parameters[i].direction == VX_OUTPUT || node->parameters[i].direction == VX_BIDIRECTIONAL))
      |                                                                                                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~

[ 14%] Building CXX object amd_openvx/openvx/CMakeFiles/openvx.dir/api/vxu.cpp.o
/home/kiriti/develop/mivisionx-kiriti/MIVisionX/amd_openvx/openvx/ago/ago_interface.cpp:2297:101: warning: result of comparison of constant 2 with expression of type 'vx_direction_e' is always false [-Wtautological-constant-out-of-range-compare]
 2297 |                         (node->parameters[i].direction == VX_INPUT || node->parameters[i].direction == VX_BIDIRECTIONAL))
      |                                                                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~
/home/kiriti/develop/mivisionx-kiriti/MIVisionX/amd_openvx/openvx/ago/ago_interface.cpp:2376:105: warning: result of comparison of constant 2 with expression of type 'vx_direction_e' is always false [-Wtautological-constant-out-of-range-compare]
 2376 |                             (node->parameters[i].direction == VX_INPUT || node->parameters[i].direction == VX_BIDIRECTIONAL))
      |                                                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~
/home/kiriti/develop/mivisionx-kiriti/MIVisionX/amd_openvx/openvx/ago/ago_interface.cpp:2405:113: warning: result of comparison of constant 2 with expression of type 'vx_direction_e' is always false [-Wtautological-constant-out-of-range-compare]
 2405 |                         if (data && (node->parameters[i].direction == VX_INPUT || node->parameters[i].direction == VX_BIDIRECTIONAL)) {
      |                                                                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~
/home/kiriti/develop/mivisionx-kiriti/MIVisionX/amd_openvx/openvx/ago/ago_interface.cpp:2461:106: warning: result of comparison of constant 2 with expression of type 'vx_direction_e' is always false [-Wtautological-constant-out-of-range-compare]
 2461 |                             (node->parameters[i].direction == VX_OUTPUT || node->parameters[i].direction == VX_BIDIRECTIONAL))
      |             
```

### Operating System

ALL

### CPU

ALL

### GPU

AMD Instinct MI300

### ROCm Version

ROCm 6.0.0

### ROCm Component

MIVisionX

**Comment by hansely** [comment]
Fixed by #1459

**Comment by kiritigowda** [comment]
@hansely reopen new pr
